src/Arcus.WebApi.Security/Authentication/Certificates/CertificateAuthenticationConfigBuilder.cs

```diff
@@ -43,8 +45,14 @@ public CertificateAuthenticationConfigBuilder WithSubject(X509ValidationLocation
         /// <exception cref="ArgumentException">Thrown when the <paramref name="configuredKey"/> is blank.</exception>
         public CertificateAuthenticationConfigBuilder WithSubject(IX509ValidationLocation location, string configuredKey)
         {
-            Guard.NotNull(location, nameof(location), "Location implementation to retrieve the expected subject cannot be 'null'");
-            Guard.NotNullOrWhitespace(configuredKey, nameof(configuredKey), "Configured key to retrieve expected subject cannot be blank");
+            if (location is null)
+            {
+                throw new ArgumentNullException(nameof(location), "Location implementation to retrieve the expected subject cannot be 'null'");
+            }
+            if (configuredKey is null)
```

Comment:
Should be checked for blank, not only `null`.
But you can also limit the guard checks to the places where it is actually used and not passed arround.